### ISSUE TITLE
recursion errors when running `connections/asyncio_connection`

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#5816]
When running the `asyncio` connection test I am getting an error that prints after (successful) testing: `Exception in default exception handler` ... `RecursionError: maximum recursion depth exceeded`. Between those two lines are a lot of repetitions of:
```
  File "/usr/local/opt/python3.5/lib/python3.5/asyncio/tasks.py", line 106, in _repr_info
    info.insert(2, 'wait_for=%r' % self._fut_waiter)
  File "/usr/local/opt/python3.5/lib/python3.5/asyncio/futures.py", line 194, in __repr__
    info = self._repr_info()
```

My guess is that we are doing something not-quite right somewhere. While looking for this I did run across the following, which implies that there are problem in some cases in using `yield from`: http://stackoverflow.com/a/28372548/4118575

Run the test like: `/test/rql_test/test-runner -ipy3.5 connections/asyncio_connection -v`. This is on both `v2.3.x` and `next` at the moment, and was happening before the changes I made recently to get this test passing (for real this time).

**Comment by danielmewes** [comment]
This appears to go away when taking out the `TestShutdown` test, and happens when running only the `TestShutdown` test.

**Comment by danielmewes** [comment]
A fix for this is in branch daniel_5816, and code review 3648.

**Comment by danielmewes** [comment]
Fixed in next and v2.3.x.
